src/main/java/randoop/generation/GrtStringFuzzer.java

```diff
@@ -0,0 +1,259 @@
+ */
+@SuppressWarnings("NotJavadoc")
+public final class GrtStringFuzzer extends GrtBaseFuzzer {
+
+  /* --------------------------- Singleton --------------------------- */
+  /** Singleton instance. */
+  private static final GrtStringFuzzer INSTANCE = new GrtStringFuzzer();
+
+  /**
+   * Obtain the singleton instance of {@link GrtStringFuzzer}.
+   *
+   * @return the singleton instance
+   */
+  public static GrtStringFuzzer getInstance() {
+    return INSTANCE;
+  }
+
+  /** Private constructor to enforce singleton. */
+  private GrtStringFuzzer() {
+    /* no-op */
+  }
+
+  /* ------------------------------ Constants ------------------------------ */
+  /** StringBuilder(String) constructor operation cached for reuse. */
+  private static final TypedClassOperation SB_CTOR_OP;
+
+  static {
+    try {
+      Constructor<StringBuilder> ctor = StringBuilder.class.getConstructor(String.class);
+      SB_CTOR_OP = TypedOperation.forConstructor(ctor);
+    } catch (NoSuchMethodException e) {
+      throw new AssertionError("StringBuilder constructor missing", e);
+    }
+  }
+
+  /** The starting ASCII value for printable characters. */
+  private static final int PRINTABLE_ASCII_START = 32;
+
+  /** Number of printable ASCII characters (codes 32-126 inclusive). */
+  private static final int PRINTABLE_ASCII_SPAN = 95;
+
+  /* ------------------------------- API ----------------------------------- */
+  @Override
+  public boolean canFuzz(Type type) {
+    return type.getRuntimeClass() == String.class;
+  }
+
+  @Override
+  public Sequence fuzz(Sequence sequence) {
+    try {
+      // 1) Grab the last runtime value:
+      Object lastValue = sequence.getStatement(sequence.size() - 1).getValue();
+
+      // 2) If it's null or not a String, just skip fuzzing:
+      if (!(lastValue instanceof String)) {
+        return sequence;
+      }
+      String strToFuzz = (String) lastValue;
+
+      StringFuzzingOperation strFuzzingOp = StringFuzzingOperation.random();
+
+      // Don't fuzz empty strings with REMOVE, REPLACE, or SUBSTRING if the original is empty
+      if (strToFuzz.isEmpty() && strFuzzingOp != StringFuzzingOperation.INSERT) {
+        return sequence;
+      }
+
+      List<Statement> stmts = new ArrayList<>();
+      // 1) Build StringBuilder(sb) from original string literal
+      Sequence strLitSeq = Sequence.createSequenceForPrimitive(strToFuzz);
+      Sequence sbSeq =
+          Sequence.createSequence(
+              SB_CTOR_OP, Collections.singletonList(strLitSeq), Collections.singletonList(0));
+      stmts.addAll(sbSeq.statements.toJDKList());
+
+      // 2) Extra inputs for the chosen fuzz operation (indices / chars)
+      Sequence opInputs = strFuzzingOp.inputs(strToFuzz.length());
+      stmts.addAll(opInputs.statements.toJDKList());
+
+      // 3) Method calls that perform the fuzzing operation
+      stmts.addAll(convertExecutables(strFuzzingOp.methods()));
```

Comment:
This seems a bit heavy-handed.  I think the work can be done in a single constant assignment (if you want to do the fuzzing at test generation time rather than at test execution time) or as a string concatenation (which would require a new class such as the PlusOperation class).